obspy/io/rg16/util.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+from future.utils import native_str as nstr
+
+import collections
+import copy
+import functools
+import struct
+
+import numpy as np
+
+from obspy import Trace
+
+BITMAP = np.array([1, 2, 4, 8])[::-1].reshape(4, 1)
+
+TENS = np.power(10, range(12))[::-1]
+
+
+def open_file(func):
```

Comment:
We should really really have something like this in `obspy.core`. Unfortunately it would require touching every single `.io` plugin which is a lot of work :(

Reply:
Please use the decorator package. ObsPy already depends on it and function signatures including documentation are correctly preserved when it is used. I'm actually kind of confused why documentation seems to be correct here. @megies Do you know? It is a thing with Python 3?

Reply:
Also please make this a private function so we don't buy into preserving it long-term.

Reply:
functools.wraps (in the std lib) handles copying the `__doc__` and `__name__` attributes. On python 3.4+ it also copies signature info.